pkg/kubelet/kubelet.go

```diff
@@ -2695,24 +2695,28 @@ func (kl *Kubelet) handlePodResourcesResize(pod *v1.Pod) {
 	kl.podResizeMutex.Lock()
 	defer kl.podResizeMutex.Unlock()
 	fit, updatedPod, resizeStatus := kl.canResizePod(pod)
+	if updatedPod == nil {
+		return
+	}
 	if fit {
 		// Update pod resource allocation checkpoint
 		if err := kl.statusManager.SetPodAllocation(updatedPod); err != nil {
 			//TODO(vinaykul,InPlacePodVerticalScaling): Can we recover from this in some way? Investigate
-			klog.ErrorS(err, "SetPodAllocation failed", "pod", klog.KObj(pod))
+			klog.ErrorS(err, "SetPodAllocation failed", "pod", klog.KObj(updatedPod))
+			return
 		}
-		*pod = *updatedPod
 	}
 	if resizeStatus != "" {
 		// Save resize decision to checkpoint
-		if err := kl.statusManager.SetPodResizeStatus(pod.UID, resizeStatus); err != nil {
+		if err := kl.statusManager.SetPodResizeStatus(updatedPod.UID, resizeStatus); err != nil {
 			//TODO(vinaykul,InPlacePodVerticalScaling): Can we recover from this in some way? Investigate
-			klog.ErrorS(err, "SetPodResizeStatus failed", "pod", klog.KObj(pod))
+			klog.ErrorS(err, "SetPodResizeStatus failed", "pod", klog.KObj(updatedPod))
+			return
 		}
-		pod.Status.Resize = resizeStatus
+		updatedPod.Status.Resize = resizeStatus
 	}
-	kl.podManager.UpdatePod(pod)
-	kl.statusManager.SetPodStatus(pod, pod.Status)
+	kl.podManager.UpdatePod(updatedPod)
+	kl.statusManager.SetPodStatus(updatedPod, updatedPod.Status)
```

Comment:
I have modified the code to return updated pod, it looks like a fairly safe change for now. PTAL.
The implementation does not rely on quick propagation, but it is desired.

My initial approach in the KEP was to not do this in SyncPod at all and do this in [HandlePodUpdates coupled with event-based retries for Deferred resizes](https://github.com/kubernetes/enhancements/pull/686#issuecomment-490629494) (e.g. revisit Deferred resizes upon deletion. IMHO it makes sense here because there is fair chance a resize try will be successful upon after other resource-using pods departed the node vs periodic retries). IIRC there were concerns about the reliability and it was conventional/accepted for K8s to retry even when it may not change the result. We also discussed this point [during early reviews](https://github.com/kubernetes/kubernetes/pull/102884#discussion_r663160060) but opted to have a beta TODO to focus on this change, tracking issue https://github.com/kubernetes/kubernetes/issues/109547

What are your thoughts on my initial suggested approach (HandlePodUpdates + event based retries)?

Reply:
Filed https://github.com/kubernetes/kubernetes/issues/116826 for addressing in the next version

Reply:
> That is the root of my concern here is that you can't use podManager as a store of temporary state (the kubelet uses it as a cache of upstream sources of truth). If you need to create a store of truth for "changes that have been requested AND accepted by the kubelet" we need to create a new map to hold them, which probably involves designing this into the state model of the kubelet as I mentioned.
> 

Yes, I plan to revisit computePodResizeAction for this purpose. Another possibility is to borrow from the [original implementation of the initially approved KEP](https://github.com/kubernetes/kubernetes/pull/92127) and patch the PodStatus and proceed only upon successful patch but that has the downside of an additional call to the API server, which could get rate-limited.